src/test/java/com/kainos/ea/service/RolesServiceTest.java

```diff
@@ -45,4 +47,30 @@ void getAllRoles_shouldThrowException_whenDaoThrowsException() throws SQLExcepti
         assertThrows(SQLException.class, () -> rolesService.getAllRoles());
     }
 
+    @Test
+    void getAllRoles_shouldReturnSpecification_whenDaoReturnsSpecification() throws SQLException, DatabaseConnectionException, IOException, RoleNotExistException {
```

Comment:
I think you can remove IOException from these method signatures as I don't see it being thrown by your dao

Reply:
there are used in databaseConnector.getConnection(), are they need to be deleted because without IOException it would not work?